panels/sep_commands.py

```diff
@@ -0,0 +1,172 @@
+import uuid
+
+from panels.models import *
+
+
+# =============================================================================
+# DEVELOPMENT TOOLS - DEVELOPMENT TOOLS - DEVELOPMENT TOOLS - DEVELOPMENT TOOLS
+# vvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvvv
+
+from sqlalchemy.engine.default import DefaultDialect
+from sqlalchemy.sql.sqltypes import String, DateTime, NullType
+
+# python2/3 compatible.
+PY3 = str is not bytes
+text = str if PY3 else unicode
+int_type = int if PY3 else (int, long)
+str_type = str if PY3 else (str, unicode)
+
+
+class StringLiteral(String):
+    """Teach SA how to literalize various things."""
+    def literal_processor(self, dialect):
+        super_processor = super(StringLiteral, self).literal_processor(dialect)
+
+        def process(value):
+            if isinstance(value, int_type):
+                return text(value)
+            if not isinstance(value, str_type):
+                value = text(value)
+            result = super_processor(value)
+            if isinstance(result, bytes):
+                result = result.decode(dialect.encoding)
+            return result
+        return process
+
+
+class LiteralDialect(DefaultDialect):
+    colspecs = {
+        # prevent various encoding explosions
+        String: StringLiteral,
+        # teach SA about how to literalize a datetime
+        DateTime: StringLiteral,
+        # don't format py2 long integers to NULL
+        NullType: StringLiteral,
+    }
+
+RE_INTERVAL = re.compile(r'\'interval "(\d+) seconds"\'')
+
+
+def literalquery(statement):
+    """
+    NOTE: This is entirely insecure. DO NOT execute the resulting strings.
+
+    USAGE: literalquery(session.query(Attendee))
+    """
+    import sqlalchemy.orm
+    if isinstance(statement, sqlalchemy.orm.Query):
+        statement = statement.statement
+    s = statement.compile(
+        dialect=LiteralDialect(),
+        compile_kwargs={'literal_binds': True},
+    ).string
+    return RE_INTERVAL.sub(r"interval '\1 seconds'", s)
+
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# DEVELOPMENT TOOLS - DEVELOPMENT TOOLS - DEVELOPMENT TOOLS - DEVELOPMENT TOOLS
+# =============================================================================
+
+
+if c.DEV_BOX:
+    @entry_point
```

Comment:
Huh.  I'm not sure why it's never occurred to me to wrap an ``@entry_point`` inside this kind of if statement.  I've always done it in reverse, but this seems better.